cdap-api/src/main/java/co/cask/cdap/api/common/Throwable.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.api.common;
+
+/**
+ * Carries {@link java.lang.Throwable} information for serialization and deserialization purpose.
+ */
+public interface Throwable {
```

Comment:
I think it is very confusing to name this Throwable, same as java.lang.Throwable. In code (where you don't see the imports) it looks like a regular throwable. And you might catch(Throwable t) and assume that that gets all Java throwables but it only gets the CDAP throwable. Also, this will lead to auto-completion errors, developers who want a java.lang.Throwable may accidentally import CDAP's Throwable. 

Please use different class name